### DISCUSSION TITLE
how much should I document?

### DISCUSSION THREAD

**Discussion by nima1313** [codewars/codewars.com#2865] [General]
hey!
I've just started trying to write clean and understandable codes and one of the aspects of that is documenting and commenting.
I was wondering how much documenting is too little or too much.

should I write what is the purpose of the main parts of the code or should it be obvious by just looking at the code alone?

**Comment by okpalan2** [comment]
Code shows/tell us how and comment tell us why we wrote the code in the first place.

**Comment by JohnMeyerhoff** [comment]
Hi,
@okpalan is right! 
If you are working on a feature and your method is already named ``public String getUserEmail(User user)`` you do not
have to add a comment explaining the signature or how the method works but it would be nice to see a short info on why you might _only_ want the email from a user that you already have a reference to. It might even help in understanding that you do not need some parts of your codebase or can remove them in refactoring.
If you keep in mind that the next person to read your comment will most likely be yourself a month from now or a team member in the future it might lead to improved focus on what is and is not important.
